src/App.tsx

```diff
@@ -1,27 +1,94 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Loader } from './Loader';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum FilterOptions {
+  ALL = 'all-button',
+  FIRST5 = 'first-five-button',
+  RED = 'red-button',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+type ButtonNames = {
+  [key in FilterOptions]: string;
+};
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+const buttonNames: ButtonNames = {
+  [FilterOptions.ALL]: 'Load all goods',
+  [FilterOptions.FIRST5]: 'Load 5 first goods',
+  [FilterOptions.RED]: 'Load red goods',
+};
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [filterOptions, setFilterOptions] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const getGoods = async (promise: Promise<Good[]>): Promise<void> => {
+    setIsLoading(true);
+    setErrorText('');
+    try {
+      const goodsFromServer = await promise;
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      setGoods(goodsFromServer);
+    } catch (error) {
+      setErrorText(String(error));
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleButtonClick = useCallback((sort: FilterOptions) => {
+    if (filterOptions === sort) {
+      return;
+    }
+
+    setFilterOptions(sort);
+
+    switch (sort) {
+      case FilterOptions.ALL:
+        getGoods(getAll());
+        break;
+      case FilterOptions.FIRST5:
+        getGoods(get5First());
+        break;
+      case FilterOptions.RED:
+        getGoods(getRedGoods());
+        break;
+      default:
+        break;
+    }
+  }, [filterOptions]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <div className="field is-grouped">
+        {Object.values(FilterOptions).map((currentSortType) => (
```

Comment:
Add a key prop to each button in the map function, using the currentSortType as the key.